x-pack/plugins/cloud_security_posture/public/pages/findings/findings_flyout/overview_tab.tsx

```diff
@@ -99,6 +116,40 @@ const getEvidenceList = ({ result }: CspFinding) =>
   ].filter(Boolean) as EuiDescriptionListProps['listItems'];
 
 export const OverviewTab = ({ data }: { data: CspFinding }) => {
+  const {
+    services: { discover },
+    notifications: { toasts },
+  } = useKibana();
+  const latestFindingsDataView = useLatestFindingsDataView();
+
+  const navigateToIndex = useCallback(async () => {
+    try {
+      // both cases should not happen, data view is loaded beforehand on findings page, this is mainly to discriminate and as a precaution
+      if (!discover.locator || !latestFindingsDataView.data) {
+        throw new Error();
+      }
+
+      await discover.locator.navigate({
+        indexPatternId: latestFindingsDataView.data.id,
+        timeRange: {
+          to: 'now',
+          from: 'now-15m',
+          mode: 'relative',
+        },
+      });
+    } catch (err) {
+      toasts.danger({
+        title: i18n.translate('xpack.csp.findings.findingsFlyout.overviewTab.indexLinkErrorTitle', {
+          defaultMessage: 'Error',
+        }),
+        body: i18n.translate(
+          'xpack.csp.findings.findingsFlyout.overviewTab.indexLinkErrorMessage',
+          { defaultMessage: 'Index link not found' }
+        ),
+      });
```

Comment:
we could probably call these directly when the services we need aren't available 

and for handling the unlikely case where `navigate` throws, we could pipe `err` to `toasts.addError(err)` 

currently we're showing a single error for 2 types: missing deps and failed navigation (both unlikely)

Reply:
as mentioned above, this is the ux team/technical writing team choice. no matter the case, the same error message is displayed.

Reply:
this seems wrong. UX/tech writes shouldn't dictate how code is written. 

we're throwing an `Error` without a message and then using that caught errors for 2 different scenarios which are very easily handled each on its own, as described above.